src/variable_rate.jl

```diff
@@ -334,16 +340,33 @@ function configure_jump_problem(prob, vr_aggregator::VR_Direct, jumps, cvrjs;
     return new_prob, variable_jump_callback, cont_agg
 end
 
-function total_variable_rate(vjumps, u, p, t, cur_rates, idx=1, prev_rate = zero(t))
-    if idx > length(cur_rates)
-        return prev_rate
-    end
-    @inbounds begin
-        new_rate = vjumps[idx].rate(u, p, t)
-        sum_rate = add_fast(new_rate, prev_rate)
-        cur_rates[idx] = sum_rate
-        return total_variable_rate(vjumps, u, p, t, cur_rates, idx + 1, sum_rate)
+@inline function fill_curr_rates(u, p, t, curr_rates, idx, rate, rates...)
+    @inbounds curr_rates[idx] = rate(u, p, t)
+    idx += 1
+    fill_curr_rates(u, p, t, curr_rates, idx, rates...)
+end
+
+@inline function fill_curr_rates(u, p, t, curr_rates, idx, rate)
+    @inbounds curr_rates[idx] = rate(u, p, t)
+    nothing
+end
```

Comment:
```suggestion
@inline function cumsum_rates!(curr_rates, u, p, t, rates)
    cur_sum = zero(eltype(curr_rates))
    cumsum_rates!(curr_rates, u, p, t, 1, cur_sum, rates...)
end

@inline function cumsum_rates!(curr_rates, u, p, t, idx, cur_sum, rate, rates...)
    new_sum = cur_sum + rate(u, p, t)
    @inbounds curr_rates[idx] = new_sum
    idx += 1        
    cumsum_rates!(curr_rates, u, p, t, idx, new_sum, rates...)
end

@inline function cumsum_rates!(curr_rates, u, p, t, idx, cur_sum, rate)
    @inbounds curr_rates[idx] = cur_sum + rate(u, p, t)
    nothing
end
```
Just calculate the cumulative sum directly.